src/Libraries/Microsoft.Extensions.DataIngestion/Processors/ClassificationEnricher.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.AI;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.DataIngestion;
+
+/// <summary>
+/// Enriches document chunks with a classification label based on their content.
+/// </summary>
+/// <remarks>This class uses a chat-based language model to analyze the content of document chunks and assign a
+/// single, most relevant classification label. The classification is performed using a predefined set of classes, with
+/// an optional fallback class for cases where no suitable classification can be determined.</remarks>
+public sealed class ClassificationEnricher : IngestionChunkProcessor<string>
+{
+    private readonly IChatClient _chatClient;
+    private readonly ChatOptions? _chatOptions;
+    private readonly TextContent _request;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ClassificationEnricher"/> class.
+    /// </summary>
+    /// <param name="chatClient">The chat client used for classification.</param>
+    /// <param name="predefinedClasses">The set of predefined classification classes.</param>
+    /// <param name="chatOptions">Options for the chat client.</param>
+    /// <param name="fallbackClass">The fallback class to use when no suitable classification is found. When not provided, it defaults to "Unknown".</param>
+    public ClassificationEnricher(IChatClient chatClient, ReadOnlySpan<string> predefinedClasses,
+        ChatOptions? chatOptions = null, string? fallbackClass = null)
+    {
+        if (predefinedClasses.Length == 0)
+        {
+            Throw.ArgumentException(nameof(predefinedClasses), "Predefined classes must be provided.");
+        }
+
+        _chatClient = Throw.IfNull(chatClient);
+        _chatOptions = chatOptions;
+        _request = CreateLlmRequest(predefinedClasses, string.IsNullOrEmpty(fallbackClass) ? "Unknown" : fallbackClass!);
+    }
+
+    /// <summary>
+    /// Gets the metadata key used to store the classification.
+    /// </summary>
+    public static string MetadataKey => "classification";
+
+    /// <inheritdoc />
+    public override async IAsyncEnumerable<IngestionChunk<string>> ProcessAsync(IAsyncEnumerable<IngestionChunk<string>> chunks,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(chunks);
+
+        await foreach (IngestionChunk<string> chunk in chunks.WithCancellation(cancellationToken))
+        {
+            var response = await _chatClient.GetResponseAsync(
+            [
+                new(ChatRole.User,
+                [
+                    _request,
```

Comment:
Should this _request be in ChatOptions.Instructions or part of a separate ChatRole.System message?